include/extractor/guidance/turn_handler.hpp

```diff
@@ -68,8 +85,7 @@ class TurnHandler : public IntersectionHandler
     handleDistinctConflict(const EdgeID via_edge, ConnectedRoad &left, ConnectedRoad &right) const;
 
     // Classification
-    std::pair<std::size_t, std::size_t> findFork(const EdgeID via_edge,
-                                                 const Intersection &intersection) const;
+    boost::optional<Fork> findFork(const EdgeID via_edge, Intersection &intersection) const;
```

Comment:
good point. I also renamed `findLeftAndRightmostForkCandidates` to `findForkCandidatesByGeometry` which is hopefully a better name because we're just looking at angles and the geometric relations between the roads here.